File: llm_outputs/codellama13b/GuruSamples/csharp/extract-method_isolate-switch_refactored.cs
class Order
{
  //...

  public double calculateTotal()
  {
    double total = 0;
    foreach (Product product in getProducts())
    {
      total += product.quantity * product.price;
    }

    // Apply regional discounts.
    total *= getDiscountFactor(user.getCountry());

    return total;
  }

  private double getDiscountFactor(string country)
  {
    switch (country)
    {
      case "US": return 0.85;
      case "RU": return 0.75;
      case "CN": return 0.9;
      //...
    }
  }
}

File: llm_outputs/codellama7b/GuruSamples/php/extract-method_isolate-switch_refactored.php
<?php
class Order {
  //...

  public function calculateTotal() {
    $total = 0;
    foreach ($this->getProducts() as $product) {
      $total += $product->quantity * $product->price;
    }

    // Apply regional discounts.
    $total *= $this->getDiscountFactor();

    return $total;
  }

  private function getDiscountFactor() {
    switch ($this->user->getCountry()) {
      case "US": return 0.85;
      case "RU": return 0.75;
      case "CN": return 0.9;
      //...
    }
  }
}